diff --git a/python-openstackclient/openstackclient/common/configuration.py b/python-openstackclient/openstackclient/common/configuration.py
index 49ef0e0..0de8923 100644
--- a/python-openstackclient/openstackclient/common/configuration.py
+++ b/python-openstackclient/openstackclient/common/configuration.py
@@ -63,4 +63,7 @@ class ShowConfiguration(command.ShowOne):
                 value = REDACTED
             info['auth.' + key] = value
 
+        if parsed_args.mask:
+            info['password'] = REDACTED
+
         return zip(*sorted(info.items()))
diff --git a/python-openstackclient/openstackclient/tests/unit/common/test_configuration.py b/python-openstackclient/openstackclient/tests/unit/common/test_configuration.py
index bdd3deb..b6bc4e9 100644
--- a/python-openstackclient/openstackclient/tests/unit/common/test_configuration.py
+++ b/python-openstackclient/openstackclient/tests/unit/common/test_configuration.py
@@ -25,6 +25,7 @@ class TestConfiguration(utils.TestCommand):
         'auth.token',
         'auth.username',
         'identity_api_version',
+        'password',
         'region',
     )
     datalist = (
@@ -32,6 +33,7 @@ class TestConfiguration(utils.TestCommand):
         configuration.REDACTED,
         fakes.USERNAME,
         fakes.VERSION,
+        configuration.REDACTED,
         fakes.REGION_NAME,
     )
 
@@ -59,9 +61,16 @@ class TestConfiguration(utils.TestCommand):
         cmd = configuration.ShowConfiguration(self.app, None)
         parsed_args = self.check_parser(cmd, arglist, verifylist)
 
-        columns, data = cmd.take_action(parsed_args)
+        col, data = cmd.take_action(parsed_args)
 
-        self.assertEqual(self.columns, columns)
+        columns = (
+            'auth.password',
+            'auth.token',
+            'auth.username',
+            'identity_api_version',
+            'region',
+        )
+        self.assertEqual(columns, col)
         datalist = (
             fakes.PASSWORD,
             fakes.AUTH_TOKEN,
